apps/server/src/modules/provisioning/strategy/oidc/service/oidc-provisioning.service.spec.ts

```diff
@@ -342,6 +344,201 @@ describe('OidcProvisioningService', () => {
 	});
 
 	describe('provisionExternalGroup', () => {
+		describe('when group membership of user has not changed', () => {
+			const setup = () => {
+				const systemId = 'systemId';
+				const externalUserId = 'externalUserId';
+				const role: RoleReference = roleFactory.buildWithId();
+				const user: UserDO = userDoFactory.buildWithId({ roles: [role], externalId: externalUserId });
+
+				const existingGroups: Group[] = groupFactory.buildList(2, {
+					users: [{ userId: user.id as string, roleId: role.id }],
+				});
+
+				const firstExternalGroup: ExternalGroupDto = externalGroupDtoFactory.build({
+					externalId: existingGroups[0].externalSource?.externalId,
+					users: [{ externalUserId, roleName: role.name }],
+				});
+				const secondExternalGroup: ExternalGroupDto = externalGroupDtoFactory.build({
+					externalId: existingGroups[1].externalSource?.externalId,
+					users: [{ externalUserId, roleName: role.name }],
+				});
+				const externalGroups: ExternalGroupDto[] = [firstExternalGroup, secondExternalGroup];
+
+				userService.findByExternalId.mockResolvedValue(user);
+				groupService.findByUser.mockResolvedValue(existingGroups);
+
+				return {
+					externalGroups,
+					systemId,
+					externalUserId,
+				};
+			};
+
+			it('should not call groupService.save', async () => {
```

Comment:
I dont know if the description is correct. Please look into the test structure docu. 
Maybe we should describe the function like e.g. should not save the groups